src/main/java/org/folio/service/orders/PurchaseOrderLineService.java

```diff
@@ -453,5 +458,41 @@ public Future<Void> validateAndNormalizeISBN(List<CompositePoLine> compositePoLi
           .mapEmpty();
       });
   }
+
+  public Future<Void> validateAndNormalizeISBNAndProductType(List<CompositePoLine> compositePoLines, RequestContext requestContext) {
+    var filteredCompLines = compositePoLines.stream()
+      .filter(HelperUtils::isProductIdsExist)
+      .toList();
+
+    if (filteredCompLines.isEmpty()) {
+      return Future.succeededFuture();
+    }
+
+    return inventoryCache.getInvalidISBNProductTypeId(requestContext)
+      .compose(invalidIsbnTypeId -> inventoryCache.getISBNProductTypeId(requestContext)
+        .compose(isbnTypeId -> {
+          var setOfProductIds = buildSetOfProductIdsFromCompositePoLines(filteredCompLines, isbnTypeId);
+          return org.folio.service.orders.utils.HelperUtils.executeWithSemaphores(setOfProductIds,
+              productId -> inventoryCache.convertToISBN13(extractProductId(productId), requestContext)
+                .map(normalizedId -> Map.entry(productId, normalizedId))
```

Comment:
I think this code can be extracted into separate method:
`.compose(invalidIsbnTypeId -> inventoryCache.getISBNProductTypeId(requestContext)
        .compose(isbnTypeId -> {
          var setOfProductIds = buildSetOfProductIdsFromCompositePoLines(filteredCompLines, isbnTypeId);
          return org.folio.service.orders.utils.HelperUtils.executeWithSemaphores(setOfProductIds,
              productId -> inventoryCache.convertToISBN13(extractProductId(productId), requestContext)
                .map(normalizedId -> Map.entry(productId, normalizedId))`